src/main/java/org/broadinstitute/hellbender/utils/read/GoogleGenomicsReadToGATKReadAdapter.java

```diff
@@ -0,0 +1,664 @@
+import com.google.api.services.genomics.model.Position;
+import com.google.api.services.genomics.model.Read;
+import com.google.cloud.dataflow.sdk.coders.DelegateCoder;
+import com.google.cloud.dataflow.sdk.coders.KvCoder;
+import com.google.cloud.dataflow.sdk.values.KV;
+import com.google.cloud.genomics.dataflow.coders.GenericJsonCoder;
+import com.google.cloud.genomics.gatk.common.GenomicsConverter;
+import htsjdk.samtools.*;
+import htsjdk.samtools.util.Locatable;
+import htsjdk.samtools.util.StringUtil;
+import org.broadinstitute.hellbender.engine.dataflow.coders.UUIDCoder;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+
+import java.util.*;
+
+public final class GoogleGenomicsReadToGATKReadAdapter implements MutableGATKRead {
+
+    private final Read genomicsRead;
+    private UUID uuid;
+
+    public GoogleGenomicsReadToGATKReadAdapter( final Read genomicsRead ) {
+        this.genomicsRead = genomicsRead;
+        this.uuid = UUID.randomUUID();
+    }
+
+    /**
+     * Dataflow coder for this adapter class
+     */
+    public static final DelegateCoder<GATKRead, KV<UUID, Read>> CODER =
+        DelegateCoder.of(
+
+            KvCoder.of(UUIDCoder.CODER, GenericJsonCoder.of(Read.class)),
+                    new DelegateCoder.CodingFunction<GATKRead, KV<UUID, Read>>() {
+                        private static final long serialVersionUID = 1l;
+
+                        @Override
+                        public KV<UUID, Read> apply( GATKRead gatkRead ) throws Exception {
+                            return KV.of(gatkRead.getUUID(), gatkRead.convertToGoogleGenomicsRead());
+                        }
+                    },
+                    new DelegateCoder.CodingFunction<KV<UUID, Read>, GATKRead>() {
+                        private static final long serialVersionUID = 1l;
+
+                        @Override
+                        public GATKRead apply( KV<UUID, Read> read ) throws Exception {
+                            final GoogleGenomicsReadToGATKReadAdapter adapter = new GoogleGenomicsReadToGATKReadAdapter(read.getValue());
+                            adapter.setUUID(read.getKey());
+                            return adapter;
+                        }
+                    }
+            );
+
+    private static <T> T assertFieldValueNotNull( final T fieldValue, final String fieldName ) {
+        if ( fieldValue == null ) {
+            throw new GATKException.MissingReadField(fieldName);
+        }
+        return fieldValue;
+    }
+
+    private void assertHasAlignment() {
+        assertFieldValueNotNull(genomicsRead.getAlignment(), "alignment");
+    }
+
+    private void assertHasPosition() {
+        assertHasAlignment();
+        assertFieldValueNotNull(genomicsRead.getAlignment().getPosition(), "position");
+    }
+
+    @Override
+    public UUID getUUID() {
+        return uuid;
+    }
+
+    public void setUUID( final UUID uuid ) {
+        this.uuid = uuid;
```

Comment:
I've killed this method, replacing it with a private constructor invoked by the nested coder class.